service-templates/apigw-lambda-svc/v1/pipeline_infrastructure/main.tf

```diff
@@ -1,393 +1,38 @@
-resource "aws_s3_bucket" "function_bucket" {
-  bucket_prefix = "function-bucket"
-}
+################################################################################
+# Terraform Backend & Provider
+################################################################################
 
-resource "aws_s3_bucket_server_side_encryption_configuration" "aes256" {
-  bucket = aws_s3_bucket.function_bucket.id
+terraform {
+  required_version = ">= 0.13.1"
 
-  rule {
-    apply_server_side_encryption_by_default {
-      sse_algorithm = "AES256"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
     }
   }
-}
 
-resource "aws_s3_bucket_policy" "function_bucket_policy" {
-  count  = var.pipeline.inputs.environment_account_ids != "" ? 1 : 0
-  policy = data.aws_iam_policy_document.function_bucket_policy_document.json
```

Comment:
when using the data source, I usually name the data source the same as the resource name. The only exception are the assume role policy documents, in that case I simply append `_assume` to the end of the name. This keeps things logically organized